athenahealth/appointment_status.go

```diff
@@ -0,0 +1,37 @@
+package athenahealth
+
+// AppointmentStatus is derived from https://docs.athenahealth.com/api/api-ref/appointment#Book-appointment
+// The athenaNet appointment status. There are several possible statuses.
+// x=cancelled
+// f=future (It can include appointments where were never checked in, even if the appointment date is in the past. It is up to a practice to cancel appointments as a no show when appropriate to do so.)
+// o=open
+// 2=checked in
+// 3=checked out
+// 4=charge entered (i.e. a past appointment)
+type AppointmentStatus string
+
+const (
+	AppointmentStatusCancelled     AppointmentStatus = "x"
+	AppointmentStatusChargeEntered AppointmentStatus = "4"
+	AppointmentStatusCheckedIn     AppointmentStatus = "2"
+	AppointmentStatusCheckedOut    AppointmentStatus = "3"
+	AppointmentStatusFuture        AppointmentStatus = "f"
+	AppointmentStatusOpen          AppointmentStatus = "o"
+)
+
+var appointmentStatuses = map[AppointmentStatus]bool{
+	AppointmentStatusCancelled:     true,
+	AppointmentStatusChargeEntered: true,
+	AppointmentStatusCheckedIn:     true,
+	AppointmentStatusCheckedOut:    true,
+	AppointmentStatusFuture:        true,
+	AppointmentStatusOpen:          true,
+}
+
+func (as AppointmentStatus) String() string {
+	return string(as)
+}
+
+func (as AppointmentStatus) Valid() bool {
+	return appointmentStatuses[as]
```

Comment:
With the change to using empty `struct{}` as the map value, please also use the idiomatic 'key exists' syntax `_, ok := map[key]`. `false` is not an appropriate value in this map and shouldn't affect this behavior.